### DISCUSSION TITLE
Streaming requests with the fetch API

### DISCUSSION THREAD

**Discussion by guest271314** [aio-libs/aiohttp#5956] [Q&A] [answered]
Does aio-libs and Python in general support `fetch()` with `ReadableStream` set to `body` https://web.dev/fetch-upload-streaming/#demo?

**Comment by webknjaz** [comment, marked as answer]
Fetch API is a concept from the JS world. It doesn't really exist as a Python standard. But yes, streaming is supported: https://docs.aiohttp.org/en/v3.7.4.post0/client_quickstart.html#streaming-response-content.

**Reply by guest271314** [reply]
This is what I have so far

```
#!/usr/bin/env python3
import aiohttp
import asyncio

chunk_size = 1024
filename = 'test.webm'
async def main():
  with open(filename, 'wb') as fd:
      while True:
          chunk = await resp.content.read(chunk_size)
          if not chunk:
            break
          fd.write(chunk)

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

I attempting to just print the content for now, though yes, also create and write to a file.

Where do I get `resp` from post such as

```
  fetch('http://localhost:1234', {
    method: 'POST',
    headers: { 'Content-Type': 'text/plain' },
    body: new ReadableStream({start(c) {c.enqueue(new Uint8Array([1])); c.close();}}),
    allowHTTP1ForStreamingUpload: true,
  })
   .then((r) => r.text())
  .then(console.log)
```

?

**Comment by guest271314** [comment]
I am currently on a machine with minimal disk space which prohibits installing `npm` and `nodejs` just to run the linked fetch upload streaming demo locally. I do have Python3 installed.
